packages/btfs-http-client/examples/upload-file-via-browser/src/App.js

```diff
@@ -1,39 +1,82 @@
-/* eslint-disable no-console */
 'use strict'
 const React = require('react')
-const ipfsClient = require('ipfs-http-client')
+const btfsClient = require('../../../../btfs-http-client/src/index')
+const config = require("../config")
+
+var errorCount = 0
 
 class App extends React.Component {
+  //BTFS-1437
   constructor () {
     super()
     this.state = {
-      added_file_hash: null
+      added_file_hash: null,
+      added_session_id:  null,
+      added_session_status: null,
+      added_session_contracts: null,
+      added_status_response: null,
     }
-    this.ipfs = ipfsClient('/ip4/127.0.0.1/tcp/5001')
+
+    this.btfs = btfsClient('/ip4/127.0.0.1/tcp/5001')
 
     // bind methods
     this.captureFile = this.captureFile.bind(this)
     this.saveToIpfs = this.saveToIpfs.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
+    //bing offline signing methods
+    this.upload = this.upload.bind(this)
+    this.displayStatus = this.displayStatus.bind(this)
+    this.sign = this.sign.bind(this)
+    this.getBatch = this.getBatch.bind(this)
+    this.getUnsignedData = this.getUnsignedData.bind(this)
+
+    this.statusTimer = null
+    this.stateTimer = null
+    this.time = Date.now()
+
   }
 
-  captureFile (event) {
-    event.stopPropagation()
-    event.preventDefault()
-    if (document.getElementById('keepFilename').checked) {
-      this.saveToIpfsWithFilename(event.target.files)
-    } else {
-      this.saveToIpfs(event.target.files)
+  setTime(){
+    this.time = Date.now()
+  }
+
+  addStatus(status) {
+    let div = document.getElementById('contractStatus')
+    if (this.state.added_status_response != null){
+      div.innerHTML += "<h3>".concat(status).concat(": ").concat(this.state.added_status_response).concat("</h3>")
+    }else {
+      div.innerHTML += "<h3>".concat(status).concat("</h3>")
     }
   }
 
+  async upload(event) {
+    const response = this.btfs.upload(
+      {
+        Hash: this.state.added_file_hash,
+        TimeNonce: this.time.toString(),
+        PrivKey: config.PrivKey,
+        PeerID: config.PeerID
+      },
+      { s: `16Uiu2HAmRfbc8E4ungNn3FWqhrKVbXotRLNk8fodgpcUeUP6nw83,16Uiu2HAmRfbc8E4ungNn3FWqhrKVbXotRLNk8fodgpcUeUP6nw83` }
+    )
+    try {
+      for await (const resp of response) {
+        this.state.added_session_id = resp.ID
```

Comment:
i think you're supposed to use `this.setState({ added_session_id: resp.ID })`

Reply:
okay, will do.